lua/toggleterm/terminal.lua

```diff
@@ -280,8 +294,8 @@ function Terminal:__spawn()
     detach = 1,
     cwd = _get_dir(self.dir),
     on_exit = __handle_exit(self),
-    on_stdout = self.on_stdout,
-    on_stderr = self.on_stderr,
+    on_stdout = __handler_wrapper(self, "on_stdout"),
```

Comment:
I wonder if the semantics below would work, i.e. using the method syntax to call the `stdout` would pass in the `self` arg
```lua
function Terminal:__stdout()
  return function(...)
    self.on_stdout(self, ...)
  end
end
--- later on
on_stdout =  self:__stdout(),
```
The same would also go for `on_stderr`. It isn't massively different, but I prefer the syntax here since it's a method call which implicitly passes `self` and behaves like other method calls. Not a blocking issue if it doesn't work, just a nicety I'd prefer/can always explore later.